File: src/lib/encryption.ts
import crypto from 'crypto';

const ALGORITHM = 'aes-256-cbc';
const ENCRYPTION_KEY = process.env.ENCRYPTION_KEY || 'my-super-secret-encryption-key-32'; // Must be 32 bytes

export function encrypt(text: string): string {
  if (!text) return '';
  const iv = crypto.randomBytes(16);
  // Ensure key is exactly 32 bytes
  const key = crypto.createHash('sha256').update(ENCRYPTION_KEY).digest();
  const cipher = crypto.createCipheriv(ALGORITHM, key, iv);
  let encrypted = cipher.update(text, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return iv.toString('hex') + ':' + encrypted;
}

export function decrypt(text: string): string {
  if (!text) return '';
  try {
    const parts = text.split(':');
    if (parts.length !== 2) return text; // If not in encrypted format, return as-is
    
    const iv = Buffer.from(parts.shift() || '', 'hex');
    const encryptedText = Buffer.from(parts.join(':'), 'hex');
    const key = crypto.createHash('sha256').update(ENCRYPTION_KEY).digest();
    
    const decipher = crypto.createDecipheriv(ALGORITHM, key, iv);
    let decrypted = decipher.update(encryptedText, 'hex', 'utf8');
    decrypted += decipher.final('utf8');
    return decrypted;
  } catch (err) {
    console.error('Decryption failed:', err);
    return text;
  }
}

File: backend/utils/encryption.py
import os
import hashlib
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend
from cryptography.hazmat.primitives import padding

ALGORITHM = 'aes-256-cbc'
ENCRYPTION_KEY = os.environ.get('ENCRYPTION_KEY', 'my-super-secret-encryption-key-32')

def get_key():
    return hashlib.sha256(ENCRYPTION_KEY.encode('utf-8')).digest()

def encrypt(text: str) -> str:
    if not text:
        return ''
    key = get_key()
    iv = os.urandom(16)
    
    padder = padding.PKCS7(128).padder()
    padded_data = padder.update(text.encode('utf-8')) + padder.finalize()
    
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
    encryptor = cipher.encryptor()
    encrypted = encryptor.update(padded_data) + encryptor.finalize()
    
    return iv.hex() + ':' + encrypted.hex()

def decrypt(encrypted_text: str) -> str:
    if not encrypted_text:
        return ''
    try:
        parts = encrypted_text.split(':')
        if len(parts) != 2:
            return encrypted_text
            
        iv = bytes.fromhex(parts[0])
        encrypted_data = bytes.fromhex(parts[1])
        key = get_key()
        
        cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
        decryptor = cipher.decryptor()
        decrypted_padded = decryptor.update(encrypted_data) + decryptor.finalize()
        
        unpadder = padding.PKCS7(128).unpadder()
        decrypted = unpadder.update(decrypted_padded) + unpadder.finalize()
        
        return decrypted.decode('utf-8')
    except Exception as e:
        print(f"Decryption failed: {e}")
        return encrypted_text
